Refacto

elilette opened the pull request.

Commit:
Refacto

Commit:
feat(refacto): remove the dependency to community module with a direct implementation

nlevee commented on variables.tf:
```diff
@@ -1,9 +1,44 @@
-    description = string
-  }))
+variable "name" {
+  description = ""
```
manque la description

nlevee commented on variables.tf:
```diff
@@ -1,9 +1,44 @@
+}
+
+variable "display_name" {
+  description = ""
```
manque la description (+ les autres :-p)

nlevee reviewed (commented):
Check le standard pour le versionning : 
<https://github.com/padok-team/docs-terraform-guidelines/blob/main/terraform_versioning.md>

Commit:
improvement(refacto): simplify implementation since we're only creating 1 service account

- remove locals
 - implement members as generic

BREAKING CHANGE: remove `kubernetes_members` and replace it by generic `members`

Commit:
improvement(refacto): add service account attributes as output

Commit:
improvement(refacto): implement generic members with the same split as roles

Commit:
chore(backstage): integrate module in Backstage

Commit:
style(variables): reorder roles and set members at the end

Commit:
style(guidelines): apply IAC guidelines (default -> this)

Commit:
feat(variables): render most variables optional as those are lists

Commit:
chore(experimental): remove optional attributes features

Commit:
docs(readme): remove usage and user stories, those are now in the docs folder

Commit:
docs(readme): update terraform-docs

Commit:
feat(versions): setup versionning as edicted by padok guidelines
